gnbsim.go

```diff
@@ -70,35 +72,48 @@ func action(c *cli.Context) error {
 	summaryChan := make(chan common.InterfaceMessage)
 	result := "PASS"
 
+	var profileWaitGrp sync.WaitGroup
 	for _, profileCtx := range config.Configuration.Profiles {
 		if !profileCtx.Enable {
 			continue
 		}
-		logger.AppLog.Infoln("executing profile:", profileCtx.Name,
-			", profile type:", profileCtx.ProfileType)
-
-		go profile.ExecuteProfile(profileCtx, summaryChan)
-
-		// Waiting for execution summary from profile routine
-		msg, ok := (<-summaryChan).(*common.SummaryMessage)
-		if !ok {
-			logger.AppLog.Fatalln("Invalid Message Type")
-		}
-
-		logger.AppSummaryLog.Infoln("Profile Name:", msg.ProfileName, ", Profile Type:", msg.ProfileType)
-		logger.AppSummaryLog.Infoln("Ue's Passed:", msg.UePassedCount, ", Ue's Failed:", msg.UeFailedCount)
-
-		if msg.UeFailedCount != 0 {
-			result = "FAIL"
-		}
-
-		if len(msg.ErrorList) != 0 {
-			logger.AppSummaryLog.Infoln("Profile Errors:")
-			for _, err := range msg.ErrorList {
-				logger.AppSummaryLog.Errorln(err)
+		profileWaitGrp.Add(1)
+
+		go func(profileCtx *profctx.Profile) {
+			defer profileWaitGrp.Done()
+
+			go profile.ExecuteProfile(profileCtx, summaryChan)
+
+			select {
+			// Waiting for execution summary from profile routine
+			case m, ok := <-summaryChan:
+				if !ok {
```

Comment:
This check should be after line 93
Currently this will simply check if the channel is closed

Reply:
I would rather just get rid of this log all together.

Reply:
I think we need to check errors at both places